### PR TITLE
feat(select): add transparent variant (FE-2630)

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour
New variant of the Select Component to allow it to be displayed as transparent and right aligned.

image

image

### Current behaviour
Select Component could be rendered with standard input background and border, and it's text is left aligned in relation to the dropdown icon.

### Checklist
- [x] Release notes (Conventional Commits) 
- [x] Unit tests
- [x] Cypress automation tests
- [x] Storybook added or updated
~~- [ ] Theme support~~
~~- [ ] Typescript `d.ts` file added or updated~~

### Additional context
Required as part of Address Fieldset.

### Testing instructions
* npm start
* Go to http://localhost:9001/?path=/story/experimental-select--default
* Check the `transparent` knob

### CODE REVIEW

**Comment by lukeemmerson** [comment]
@grabkowski i’ve mocked up a HTML example here: http://visualdesigner.co.uk/95268538c23a4c9391b113fc408ab44a/ If you click new client then scroll to the address area. I’ve also mocked up the dropdown state - can we have the font-weight to be 900 for the text in the select field and the dropdown text to also be right aligned like in the HTML prototype.

File: cypress/support/step-definitions/select-steps.js [hunk 1]
```diff
@@ -92,3 +92,26 @@ When('Type {string} text into input and select the value', (text) => {
 When('Type {string} text into input and select the value into iFrame', (text) => {
   selectInputNoIframe().type(`${text}{downarrow}{enter}`);
 });
+
+Then('Select is transparent', () => {
+  select().parent().should('have.css', 'background-color', 'rgba(0, 0, 0, 0)')
+    .and('have.css', 'border-left-color', 'rgba(0, 0, 0, 0.85)')
```
Comment (DlgSHi) [inline cypress/support/step-definitions/select-steps.js]: could we make a `const color = rgba(0, 0, 0, 0.85)` and use it in every `.and` ?

File: cypress/support/step-definitions/select-steps.js [hunk 2]
```diff
@@ -92,3 +92,26 @@ When('Type {string} text into input and select the value', (text) => {
...
+    .and('have.css', 'border-right-color', 'rgba(0, 0, 0, 0.85)')
+    .and('have.css', 'border-top-color', 'rgba(0, 0, 0, 0.85)');
+});
+
+Then('Select is not transparent', () => {
+  select().parent().should('have.css', 'background-color', 'rgb(255, 255, 255)')
+    .and('have.css', 'border-left-color', 'rgb(102, 132, 145)')
```
Comment (DlgSHi) [inline cypress/support/step-definitions/select-steps.js]: could we make a `const color = rgb(102, 132, 145` and use it in every `.and` ?

File: cypress/support/step-definitions/select-steps.js [hunk 3]
```diff
@@ -92,3 +92,26 @@ When('Type {string} text into input and select the value', (text) => {
...
+    .and('have.css', 'border-right-color', 'rgb(102, 132, 145)')
+    .and('have.css', 'border-top-color', 'rgb(102, 132, 145)');
+});
+
+Then('Select placeholder align on preview is set to {string}', (direction) => {
+  if (direction === 'left') {
+    // left is default property that's why it's absent inside class
+    select().should('have.css', 'text-align', `${direction}`);
```
Comment (DlgSHi) [inline cypress/support/step-definitions/select-steps.js]: what for are passing here parameter using string interpolation ? we just could pass it there

File: cypress/support/step-definitions/select-steps.js [hunk 4]
```diff
@@ -92,3 +92,26 @@ When('Type {string} text into input and select the value', (text) => {
...
+  } else {
+    select().should('have.css', 'text-align', `${direction}`);
```
Comment (DlgSHi) [inline cypress/support/step-definitions/select-steps.js]: here should be probably 
```suggestion
    select().should('not.have.css', 'text-align', direction);
```

File: cypress/support/step-definitions/select-steps.js [hunk 5]
```diff
@@ -92,3 +92,26 @@ When('Type {string} text into input and select the value', (text) => {
...
+    .and('have.css', 'border-left-color', 'rgb(102, 132, 145)')
+    .and('have.css', 'border-right-color', 'rgb(102, 132, 145)')
+    .and('have.css', 'border-top-color', 'rgb(102, 132, 145)');
+});
+
+Then('Select placeholder align on preview is set to {string}', (direction) => {
+  if (direction === 'left') {
+    // left is default property that's why it's absent inside class
+    select().should('have.css', 'text-align', `${direction}`);
+  } else {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/select-steps.js]: what will be if we pass different value that `left` / `right` ?

File: cypress/features/regression/experimental/select.feature [hunk 6]
```diff
@@ -100,4 +100,17 @@ Feature: Experimental Select component
   Scenario: Check the change function call for Select component
     Given clear all actions in Actions Tab
     When Type "Black" text into input and select the value
-    Then change action was called in Actions Tab
\ No newline at end of file
+    Then change action was called in Actions Tab
+
+  @positive
+  Scenario: Select is transparent
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/select.feature]:
```suggestion
  Scenario: Select is transparent and has placeholder right aligned
```

File: cypress/support/step-definitions/select-steps.js [hunk 7]
```diff
@@ -92,3 +92,26 @@ When('Type {string} text into input and select the value', (text) => {
 When('Type {string} text into input and select the value into iFrame', (text) => {
   selectInputNoIframe().type(`${text}{downarrow}{enter}`);
 });
+
+Then('Select is transparent', () => {
+  select().parent().should('have.css', 'background-color', 'rgba(0, 0, 0, 0)')
```
Comment (DlgSHi) [inline cypress/support/step-definitions/select-steps.js]: font-weight should be also added

Comment (KatarzynaQA) [reply cypress/support/step-definitions/select-steps.js]: fixed

File: cypress/features/regression/experimental/select.feature (hunk 6 above)
Comment (KatarzynaQA) [reply cypress/features/regression/experimental/select.feature]: fixed

File: cypress/support/step-definitions/select-steps.js (hunk 4 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/select-steps.js]: in step I pass proper 'text-align', it's why I used `have.css`

File: cypress/support/step-definitions/select-steps.js (hunk 3 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/select-steps.js]: fixed

File: cypress/support/step-definitions/select-steps.js (hunk 2 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/select-steps.js]: fixed

File: cypress/support/step-definitions/select-steps.js (hunk 1 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/select-steps.js]: fixed

**Comment by harpalsingh** [comment]
As discussed in UX and @DlgSHi if the transparent select is active, the open menu options should still left align. The width will be handled by the developer to prevent an issue where the visible value and options are too far apart.

**Comment by grabkowski** [comment]
Hi @harpalsingh, options are right aligned per @lukeemmerson suggestion. Please look at the prototype he has provided: http://visualdesigner.co.uk/95268538c23a4c9391b113fc408ab44a/
The select is available after clicking `new client`.

**Comment by harpalsingh** [comment]
@grabkowski Yes, this is something I have discussed with @lukeemmerson and @benwilsonsage and reviewed what we do on the live app today and prototype is being updated to reflect that as well now, thanks.

**Comment by harpalsingh** [comment]
Updates all good 👍

**Pull request merged**